src/pages/sc-end-site-count/forms/EndSiteCountForm.js

```diff
@@ -0,0 +1,139 @@
+import { Grid } from '@mui/material'
+import * as yup from 'yup'
+import { useContext } from 'react'
+import FormShell from 'src/components/Shared/FormShell'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { ControlContext } from 'src/providers/ControlContext'
+import { SCRepository } from 'src/repositories/SCRepository'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import toast from 'react-hot-toast'
+
+export default function EndSiteCountForm({ _labels, access }) {
+  const { postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const { formik } = useForm({
+    initialValues: {
+      stockCountId: '',
+      siteId: '',
+      notes: '',
+      status: '',
+      statusName: ''
+    },
+    enableReinitialize: true,
+    maxAccess: access,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      stockCountId: yup.string().required(),
+      siteId: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      try {
+        const copy = { ...obj }
+        delete copy.statusName
+
+        if (copy.status === 3) {
+          await postRequest({
+            extension: SCRepository.Sites.reopen,
+            record: JSON.stringify(copy)
+          })
+          toast.success(platformLabels.Saved)
+        } else {
+          await postRequest({
+            extension: SCRepository.Sites.end,
+            record: JSON.stringify(copy)
+          })
+          
+          toast.success(platformLabels.Saved)
+        }
+
+        formik.resetForm()
+      } catch (error) {}
+    }
+  })
+
+  const actions = [
+    {
+      key: 'Lock',
+      condition: formik.values.statusName === 'Processed',
+      onClick: () => formik.handleSubmit(),
+      disabled: false
+    },
+    {
+      key: 'Unlock',
+      condition: formik.values.statusName !== 'Processed',
+      onClick: () => formik.handleSubmit(),
```

Comment:
is it better to use status =1 or 2 ?

Reply:
same

Reply:
@imtithalhalalo don't compare based on the statusName, what if we're using argus in arabic ? @Alinadimmokdad is right